pkg/controller/networkpolicy/clustergroup.go

```diff
@@ -200,3 +202,26 @@ func (n *NetworkPolicyController) syncInternalGroup(key string) error {
 	n.internalGroupStore.Update(updatedGrp)
 	return nil
 }
+
+func (n *NetworkPolicyController) GetAssociatedGroups(name, namespace string) (interface{}, error) {
+	memberKey := store.GroupMemberKeyFunc(name, namespace)
+	groups, err := n.internalGroupStore.GetByIndex(store.GroupMemberIndex, memberKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve Group associated with key %s", memberKey)
+	}
+	groupObjs := make([]*antreatypes.Group, len(groups))
+	for i, g := range groups {
+		groupObjs[i] = g.(*antreatypes.Group)
```

Comment:
Why don't return a slice of `antreatypes.Group` directly but a slice of pointers and convert it again in caller? 
Why the return value is `interface{}`?

Reply:
I think I was following another rest.go example but couldn't remember which. Refactored both query method to return specific types.

Reply:
Since you are converting `interface{}` to concrete type, you can just return a slice of `antreatypes.Group`, instead of their pointers to avoid another loop in rest.go given that it's the only caller?

Reply:
Sure.